src/transformers/models/llava_onevision/processing_llava_onevision.py

```diff
@@ -0,0 +1,275 @@
+
+            image_sizes = to_py_obj(image_inputs["image_sizes"])
+            height, width = get_image_size(to_numpy_array(image_inputs["pixel_values"][0][0]))
+            prompt_strings = []
+
+            for sample in text:
+                while self.image_token in sample:
+                    image_size = image_sizes.pop(0)
+                    orig_height, orig_width = image_size
+                    num_image_tokens = self._get_number_of_features(orig_height, orig_width, height, width)
+                    if self.vision_feature_select_strategy == "default":
+                        num_image_tokens -= 1
+
+                    sample = sample.replace(self.image_token, "<placeholder>" * num_image_tokens, 1)
+                prompt_strings.append(sample)
+            text = [sample.replace("<placeholder>", self.image_token) for sample in prompt_strings]
+
+        if videos is not None:
+            video_inputs = self.video_processor(videos, **output_kwargs["videos_kwargs"])
+
+            image_sizes = to_py_obj(video_inputs["image_sizes_videos"])
+            one_video = to_numpy_array(video_inputs["pixel_values_videos"][0])
+            height, width = get_image_size(one_video[0])
+            num_frames = one_video.shape[0]  # frame dim is always after batch dim
+            num_image_tokens = self.num_image_tokens
+            prompt_strings = []
+
+            for sample in text:
+                while self.video_token in sample:
+                    image_size_list = image_sizes.pop(0)
+                    orig_height, orig_width = image_size_list[0]
+                    num_image_tokens = self._get_number_of_features(orig_height, orig_width, height, width)
+                    if self.vision_feature_select_strategy == "default":
+                        num_image_tokens -= 1
+
+                    sample = sample.replace(self.video_token, "<placeholder>" * num_image_tokens * num_frames, 1)
+                prompt_strings.append(sample)
+            text = [sample.replace("<placeholder>", self.video_token) for sample in prompt_strings]
+
+        # Padding side can be in TextKwargs but is not accepted by the tokenizer
+        _ = output_kwargs["text_kwargs"].pop("padding_side", None)
+        text_inputs = self.tokenizer(text, **output_kwargs["text_kwargs"])
+        return BatchFeature(data={**text_inputs, **image_inputs, **video_inputs})
+
+    def _get_number_of_features(self, orig_height: int, orig_width: int, height: int, width: int) -> int:
+        image_grid_pinpoints = self.image_processor.image_grid_pinpoints
+
+        height_best_resolution, width_best_resolution = select_best_resolution(
+            [orig_height, orig_width], image_grid_pinpoints
+        )
+        scale_height, scale_width = height_best_resolution // height, width_best_resolution // width
+
+        patches_height = int(math.sqrt(self.num_image_tokens))
+        patches_width = int(math.sqrt(self.num_image_tokens))
```

Comment:
nit - we don't need to worry about shared references for an int here 

```suggestion
        patches_height = patches_width = int(math.sqrt(self.num_image_tokens))
```